tests/unit/db/account_tests.py

```diff
@@ -0,0 +1,640 @@
+        return '{0}-{1}'.format(database_name, unicode(uuid.uuid4()))
+
+    def test_constructor_with_url(self):
+        """
+        Test instantiating an account object using a URL
+        """
+        self.assertEqual(
+            self.client.cloudant_url,
+            self.url
+            )
+        self.assertEqual(self.client.encoder, json.JSONEncoder)
+        self.assertIsNone(self.client.r_session)
+
+    def test_connect(self):
+        """
+        Test connect and disconnect functionality
+        """
+        try:
+            self.client.connect()
+            self.assertIsInstance(self.client.r_session, requests.Session)
+            self.assertEqual(self.client.r_session.auth, (self.user, self.pwd))
+        finally:
+            self.client.disconnect()
+            self.assertIsNone(self.client.r_session)
+
+    def test_session(self):
+        """
+        Test getting session information
+        """
+        try:
+            self.client.connect()
+            session = self.client.session()
+            self.assertEqual(session['userCtx']['name'], self.user)
+        finally:
+            self.client.disconnect()
+
+    def test_session_cookie(self):
+        """
+        Test getting the session cookie
+        """
+        try:
+            self.client.connect()
+            self.assertIsNotNone(self.client.session_cookie())
+        finally:
+            self.client.disconnect()
+
+    def test_basic_auth_str(self):
+        """
+        Test getting the basic authentication string
+        """
+        try:
+            self.client.connect()
+            expected = 'Basic {0}'.format(
+                base64.urlsafe_b64encode("{0}:{1}".format(self.user, self.pwd))
+                )
+            self.assertEqual(
+                self.client.basic_auth_str(),
+                expected
+                )
+        finally:
+            self.client.disconnect()
+
+    def test_all_dbs(self):
+        """
+        Test getting a list of all of the databases in the account
+        """
+        try:
+            self.client.connect()
+            self.assertIsInstance(self.client.all_dbs(), list)
+        finally:
+            self.client.disconnect()
+
+    def test_create_delete_database(self):
+        """
+        Test database creation and deletion
+        """
+        try:
+            self.client.connect()
+            dbname = self.dbname()
+            # Create database
+            db = self.client.create_database(dbname)
+            self.assertTrue(db.exists())
+            # Delete database
+            self.assertIsNone(self.client.delete_database(dbname))
+            self.assertFalse(db.exists())
+        finally:
+            self.client.disconnect()
+
+    def test_create_existing_database(self):
+        """
+        Test creation of already existing database
+        """
+        try:
+            self.client.connect()
+            self.client.create_database('_replicator', throw_on_exists=True)
```

Comment:
I think this would be better creating a database then testing it. I vaguely recall that `_replicator` databases are not created in Cloudant accounts until the first replication job, so this test would fail on a new Cloudant account.